### ISSUE TITLE
Global Bindings for ldap policies

### ISSUE THREAD

**Issue by andrewhadam** [citrix/terraform-provider-citrixadc#360]
Hopefully I haven't overlooked something obvious but was excited to see the addition of ldap configuration for the provider!  I went to add it to our configuration and found the following error:

203: globalbinding {
Blocks of type "globalbinding" are not expected here

I do see you're releasing a lot of authentication updates recently but wanted to make sure this was on your radar in case it was not!  I also hunted around to see if there was an equivalent to 

https://developer-docs.citrix.com/projects/netscaler-nitro-api/en/latest/configuration/system/systemglobal_authenticationpolicy_binding/

But didn't see it yet.  Thanks!

**Comment by giorgos-nikolopoulos** [comment]
@andrewhadam 

Could you please include the configuration files and logs from the execution?

**Comment by andrewhadam** [comment]
My apologies for the delay but yep, I can definitely provide what I was trying.  Here is my scenario:

I have an authentication ldapAction called testImplementation. 
Then I have the following two lines: 

> add authentication Policy test_implmentation -rule true -action test_implementation
> bind system global test_implmentation -priority 100 -gotoPriorityExpression NEXT

The ldapAction + the Policy I can make without an issue.  The following binding, I was unable to make.  Below is the equivalent Terraform code I was working with.

``` 
resource "citrixadc_authenticationldapaction" "default_ldap_actions" {
  for_each                = var.ldap_groups
  name                    = each.value.ldapaction_name
  serverport              = lookup(var.ldap_defaults, "serverport")
  ldapbase                = lookup(var.ldap_defaults, "ldapbase")
  ldapbinddn              = lookup(var.ldap_defaults, "ldapbinddn")
  ldaploginname           = lookup(var.ldap_defaults, "ldaploginname")
  groupattrname           = lookup(var.ldap_defaults, "groupattrname")
  subattributename        = lookup(var.ldap_defaults, "subattributename")
  sectype                 = lookup(var.ldap_defaults, "sectype")
  nestedgroupextraction   = lookup(var.ldap_defaults, "nestedgroupextraction")
  groupnameidentifier     = lookup(var.ldap_defaults, "groupnameidentifier")
  groupsearchattribute    = lookup(var.ldap_defaults, "groupsearchattribute")
  groupsearchsubattribute = lookup(var.ldap_defaults, "groupsearchsubattribute")
  servername              = lookup(lookup(var.ldap_servers, var.datacenter), "servername")
  searchfilter            = each.value.searchfilter
  ldapbinddnpassword      = var.ldapbinddnpassword
}

resource "citrixadc_authenticationpolicy" "default_ldap_policies" {
  for_each = var.ldap_groups
  name     = each.value.policy_name
  rule     = true
  action   = each.value.ldapaction_name

  globalbinding {
    priority = 101
    feature = "SYSTEM"
    globalbindtype = "SYSTEM_GLOBAL"
  }
}
```

I had used the globalbinding like above in some other code for the resource "citrixadc_auditsyslogpolicy" and was hoping it might also apply here. 

Let me know if I've missed anything that would help!

**Comment by giorgos-nikolopoulos** [comment]
Currently there is no system binding implemented for the authenticationpolicy resource.

We will have to create a new resource which will utilize the systemglobal_authenticationldappolicy_binding endpoint, to implement this functionality.

**Comment by giorgos-nikolopoulos** [comment]
Fixed with #426 
Published with provider version 1.14.0
